File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();

        vector<int> res;

        for(int i = 0; i < n; i++){
            if(nums[i] != 0){
                res.push_back(nums[i]);
            }
        }

        for(int i = 0; i < res.size(); i++){
            nums[i] = res[i];
        }
        for(int i = res.size(); i < n; i++){
            nums[i] = 0;
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        n = len(nums)
        i = 0
        j = 1

        while(i < n and j < n):
            if(nums[i] == 0 and nums[j] != 0):
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
                j += 1
            elif(nums[i] == 0 and nums[j] == 0):
                j += 1    
            else:
                i += 1
                j += 1
        return nums        
    
        